common/include/villas/kernel/devices/ip_device_reader.hpp

```diff
@@ -0,0 +1,38 @@
+/* Devicetree reader which parses fpga ip devices from a filesystem path
+ *
+ * Author: Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ *
+ * SPDX-FileCopyrightText: 2023-2024 Pascal Bauer <pascal.bauer@rwth-aachen.de>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <filesystem>
+#include <stdexcept>
+#include <vector>
+
+#include <villas/kernel/devices/ip_device.hpp>
+#include <villas/utils.hpp>
+
+class IpDeviceReader {
+public:
+  std::vector<villas::kernel::devices::IpDevice>
+  read(std::filesystem::path devices_directory) const {
```

Comment:
Are you sure you are using the correct clang-format config? This looks not like other code in VILLASnode.